tachyon/zk/r1cs/groth16/prove.h

```diff
@@ -223,7 +223,7 @@ Proof<Curve> ReRandomizeProof(const VerifyingKey<Curve>& vk,
   //   B' = r₁B + r₁r₂(δG₂)
   //   C' = C + r₂A
   G1JacobianPoint ac_jacobian[2];
-  ac_jacobian[0] = proof.a() * randoms.r1.Inverse();
+  ac_jacobian[0] = proof.a() * *randoms.r1.Inverse();
```

Comment:
Please add a comment that we can be sure that this is not zero.

Reply:
From the lines 217/218 above, I assumed that `randoms.r1` would not be zero, yet `AreZeroes()` only check if both of them are zero together. I'll change line 226 to check if the inverse if invalid!